### DISCUSSION TITLE
How to extend rhythmic in tablature ?

### DISCUSSION THREAD

**Discussion by zhouhao27** [CoderLine/alphaTab#1235] [Q&A] [answered]
My display is like this:

<img width="989" alt="Screenshot 2023-07-26 at 5 52 59 PM" src="https://github.com/CoderLine/alphaTab/assets/8099731/de7e1995-798e-4675-99b0-b473a8033c8e">

How to show it like this?

<img width="796" alt="Screenshot 2023-07-26 at 5 57 49 PM" src="https://github.com/CoderLine/alphaTab/assets/8099731/acecc9ce-425d-4162-8497-1579e2ab031a">

**Comment by zhouhao27** [comment, marked as answer]
Found a way to do this:

```
const at = new alphaTab.AlphaTabApi(el, {
        file: 'https://www.alphatab.net/files/canon.gp',
        notation: {
            rhythmMode: 1
        },
        player: {
            enablePlayer: true,
            soundFont: 'https://cdn.jsdelivr.net/npm/@coderline/alphatab@alpha/dist/soundfont/sonivox.sf2',
            scrollElement: viewPort,
            scrollOffsetX: -10
        }
    });
```

**Reply by Danielku15** [reply]
Sorry for the delayed answer. AlphaTab currently always extends the lines up to the note. I would be surprised to see variant 1 in alphaTab, we don't have a feature for that (yet?). The rhythmMode just controls whether the bars between notes (e.g. two quarter notes being connected) are broken into beams/flags or not.
